File: pkg/api/types/identity.go
package types

// Identity represents a user including the claims from the auth request
type Identity struct {
	ID            string `json:"id"`
	Subject       string `json:"subject"`
	DisplayName   string `json:"displayName"`
	FullName      string `json:"fullName"`
	Email         string `json:"email"`
	EmailVerified bool `json:"emailVerified"`
}

File: frontend/packages/core-api-client/src/types/client/Identity.ts
import { DataOperation, Response } from './utils';

export type Identity = {
  id: string;
  subject: string;
  displayName: string;
  fullName: string;
  email: string;
  emailVerified: boolean;
};

export type IdentityGetRequest = { id: string };

export type IdentityGetResponse = Response<undefined, Identity>;

export interface IdentityClientDefinition {
  get: DataOperation<IdentityGetRequest, IdentityGetResponse>;
}
